std/haxe/Int64Helper.hx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C)2005-2015 Haxe Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package haxe;
+
+using haxe.Int64;
+
+import StringTools;
+
+class Int64Helper {
+
+	public static function fromString( sParam : String ) : Int64 {
+		var base = Int64.ofInt(10);
+		var current = Int64.ofInt(0);
+		var multiplier = Int64.ofInt(1);
+		var sIsNegative = false;
+
+		var s = StringTools.trim(sParam);
+		if (s.charAt(0) == "-") {
+			sIsNegative = true;
+			s = s.substring(1, s.length);
+		}
+		var len = s.length;
+
+		for (i in 0...len) {
+			var char = s.charAt(len - 1 - i);
+			var digit:Int64 = Int64.ofInt(0);
+
+			switch (char) {
```

Comment:
Haxe allows you to write this much shorter as:

```haxe
digit = Int64.ofInt(switch (char) { 
   case "0": 0
   case "1": 1
   // ...
});
```

Which removes the redundancy of having `digit` and `Int64.ofInt()` inside of every single case. This works because a switch in Haxe is an expression that "returns" a value.